include/bout/coordinates.hxx

```diff
@@ -210,6 +231,13 @@ private:
   int nz; // Size of mesh in Z. This is mesh->ngz-1
   Mesh * localmesh;
   CELL_LOC location;
+
+  /// Handles calculation of yup and ydown
+  std::unique_ptr<ParallelTransform> transform{nullptr};
+
+  /// Set the parallel (y) transform from the options file.
+  /// Usual way to create the transform object.
+  void setParallelTransform(Options* options);
```

Comment:
I was thinking it would be the usual way to create a `ParallelTransform`, but only ever be called in the `Coordinates` constructor. I guess I don't expect users to have a need to change the `ParallelTransform` once `Mesh` and `Coordinates` are first created; isn't it better to minimise the public interface? If someone really wants to do this they can use the `setParallelTransform(std::unique_ptr<ParallelTransform> pt)` method. Happy to move to public if you think it's better though, I've not got a strong opinion on it. Could also make it `protected` in case we might at some point have derived classes of `Coordinates` which need access, e.g. https://github.com/boutproject/BOUT-dev/blob/f0d55cd749587143008a770e3db6495416483a38/include/bout/coordinates.hxx#L244-L245

Should I move the method to `public`, move to `protected` or just change the comment?